### ISSUE TITLE
unused_dependency_checker false positives with _suite macros

### ISSUE THREAD

**Issue by gkossakowski** [bazelbuild/rules_scala#617]
Unused dependency checker gives false positives when _suite macros are involved. Let's consider:

```
scala_library_suite(
   name = "foo",
   srcs = ["A.scala", "B.scala"],
   deps = ["dep1", "dep2"],
)
```

`A.scala` depends on `dep1` and `B.scala` depends on `dep2`. In this scenario, both `dep1` and `dep2` will be reported as unused. It's because _suite is expanded into several targets with deps copied to each of them.

Users' expectation here is that dependency is reported as an unused only when it's not used in all srcs from the suite but now it's reported if it's unused in some source from the suite. Fixing this would require a post-processing step that takes union of deps information from all expanded targets of the suite.

**Comment by ittaiz** [comment]
Can we also think how this works for a scala_library with java and scala?
Feels related (though definitely not the same). We (Wix) would like to use
this feature later on and this will need to be a supported feature (even if
we add this later on).
@iirina @lberki,
Is there an output we can take from java_common.compile to later warn about
unused deps from skylark? Want to have an aggregate of java and scala.
